### DISCUSSION TITLE
maxit and parameter convergence

### DISCUSSION THREAD

**Discussion by emmats** [atsa-es/fish550-2023#30] [Q&A]
As we prepare to hand in our Lab 3, we are wondering what to do about setting maxit for our document. if we set it at 1000, our parameters still do not converge and it takes a really long time to knit the document since we have multiple models to go through. For the instructors who will be tasked with going through our work, do you want us to leave maxit low (~100) for ease of knitting? Or do you want us to set it higher? We understand that we should set it higher!

**Comment by mdscheuerell** [comment]
Hi @emmats. Here are a few things that should help you out:
1) You can increase the `maxit` argument;
2) You can try `method = "BFGS"` instead of the default `method = "kem"`, which is a lot faster; and
3) You can add `cache = TRUE` to the code chunk option in your `.Rmd` file to save the output and not re-run the code chunk every time.

For example, your code block could look something like this
````
```{r fit_DFA, cache = TRUE}
model_1 <- MARSS(...)
```
````

**Reply by emmats** [reply]
Thanks!
